test/orange/orangeinp/CsgTree.test.cc

```diff
@@ -22,6 +22,12 @@ namespace test
 //---------------------------------------------------------------------------//
 TEST(CsgTypes, hash)
 {
+#ifdef _MSC_VER
+    // XXX
+    GTEST_SKIP()
```

Comment:
Do we rely on the `variant_hash` for the library code to work properly?

Reply:
Nope. It just means we've got a worse hash function: we're going to have more collisions and the CSG tree construction might be a bit slower as a result.

Reply:
It was more of a "let's try to understand what the compiler/library are doing" test: if it's embedding the type in the hash, it's not necessary to do it for our hash function. If it's not, then it might be worth trying both ways.

Reply:
fair enough.